by-language/java-jooq/src/generated/java/io/crate/demo/jooq/model/DemoDatabase.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is generated by jOOQ.
+ */
+package io.crate.demo.jooq.model;
+
+
+import io.crate.demo.jooq.model.tables.Author;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.processing.Generated;
+
+import org.jooq.Catalog;
+import org.jooq.Table;
+import org.jooq.impl.SchemaImpl;
+
+
+/**
+ * This class is generated by jOOQ.
+ */
+@Generated(
+        value = {
+                "https://www.jooq.org",
+                "jOOQ version:3.17.7"
+        },
+        comments = "This class is generated by jOOQ."
+)
```

Comment:
@seut mentioned that the pseudo-generated classes should not contain comments like this.

Instead, the comments should be more clear about that they have been hand-written / derived from other examples, eventually citing the README:

https://github.com/crate/cratedb-examples/blob/eb951b9635a56a5c827516159f1c44051459fd52/by-language/java-jooq/README.rst?plain=1#L46-L52

Reply:
How would a solution look like without generated code ? Would that result in the same kind of classes ?

Reply:
@mkleen Afaik this can be seen here https://github.com/crate/cratedb-examples/blob/eb951b9635a56a5c827516159f1c44051459fd52/by-language/java-jooq/src/main/java/io/crate/demo/jooq/Application.java#L132.

Reply:
GH-11 takes a better approach on this. Instead of leaving us or any reader hanging on this detail, it uses the existing SQL DDL scripts to reflect the database schema, see _DDLDatabase: Code generation from SQL files_ [^1]. Let me know what you think about it.

As outlined within the upstream documentation, and accordingly cited within the README now, this fits well with the very reasonable scenario that your SQL DDL statements are usually maintained in form of multiple incremental migration scripts anyway, when writing an application where schema changes are important to be tracked within the source code repository. Within this scenario, there is no need to connect to an existing database for reflecting its schema.

jOOQ even promises it could cope well with CrateDB-specific extensions to the `CREATE TABLE` DDL statements, which definitively matters.

> While the script uses pretty standard SQL constructs, you may well use some vendor-specific extensions, and even [DML statements](https://www.jooq.org/doc/latest/manual/sql-building/sql-statements/) in between to set up your schema - it doesn't matter. By [ignoring unsupported content](https://www.jooq.org/doc/latest/manual/code-generation/codegen-ddl/#ignoring-unsupported-content), the [jOOQ SQL parser](https://www.jooq.org/doc/latest/manual/sql-building/sql-parser/) supports parsing everything that is representable through the jOOQ API, as well as some well known vendor-specific syntax, [and even unknown vendor-specific syntax by adding correspdonding hints].

[^1]: https://www.jooq.org/doc/latest/manual/code-generation/codegen-ddl/